dev/run-tests

```diff
@@ -17,239 +17,394 @@
 # limitations under the License.
 #
 
-# Go to the Spark project root directory
-FWDIR="$(cd "`dirname $0`"/..; pwd)"
-cd "$FWDIR"
+import os
+import re
+import shutil
+import subprocess as sp
+
+# Set the Spark project root directory
+spark_proj_root = os.path.abspath("..")
```

Comment:
The variable names and logic here are very clear, so I think we can do without all the comments in this whole starting block.